### PR TITLE
Docker for sanity

### PR DESCRIPTION
**Pull request by veblush**
Upgrading it to use:
- `Debian:10` instead of 9
- `Clang 7` instead of 5
   - `Clang 6` for `clang-tidy` to avoid a sort of breaking chance due to [the issue](https://bugs.llvm.org/show_bug.cgi?id=27628).

This is to unblock #22149 which won't work with the existing docker image.

### CODE REVIEW

**Comment by veblush** [comment]
This is interesting because the previous clang-tidy doesn't return non-zero when it fails to parse the source. But newer version of clang-tidy started to treat ([issue](https://bugs.llvm.org/show_bug.cgi?id=27628)) it as an error so we have to address this.

File: templates/tools/dockerfile/test/sanity/Dockerfile.template [hunk 1]
```diff
@@ -31,7 +33,17 @@
...
 
-  <%include file="../../clang5.include"/>
+  # Install clang & clang-format
+  RUN apt-get install -y clang clang-format
+
+  # Install clang-tidy 6.0
+  # This is because clang-tidy 7.0 started treating compiler errors as tidy errors
+  # and there are a couple of files which are not properly compiled via tidy so it
+  # should be using 6.0 version until all compilation errors are addressed.
+  RUN apt-get install -y clang-tidy-6.0
```
Comment (jtattermusch) [inline templates/tools/dockerfile/test/sanity/Dockerfile.template]: nit: in Dockerfiles, you always need to prepend apt-get install with `apt-get update && `

File: templates/tools/dockerfile/grpc_clang_format/Dockerfile.template [hunk 2]
```diff
@@ -13,10 +13,10 @@
...
   # See the License for the specific language governing permissions and
   # limitations under the License.
-  
-  <%include file="../debian_jessie_header.include"/>
-  
-  <%include file="../clang5.include"/>
+
+  FROM debian:10
+
+  RUN apt-get update && apt-get install -y clang-format
```
Comment (jtattermusch) [inline templates/tools/dockerfile/grpc_clang_format/Dockerfile.template]: probably a good idea to freeze the clang version by `clang-format-7`?

File: tools/dockerfile/grpc_clang_format/Dockerfile [hunk 3]
```diff
@@ -12,17 +12,9 @@
...
-RUN apt-get update && apt-get -y install wget xz-utils
-RUN wget http://releases.llvm.org/5.0.0/clang+llvm-5.0.0-linux-x86_64-ubuntu14.04.tar.xz
-RUN tar xf clang+llvm-5.0.0-linux-x86_64-ubuntu14.04.tar.xz
-RUN ln -s /clang+llvm-5.0.0-linux-x86_64-ubuntu14.04/bin/clang-format /usr/local/bin/clang-format
-ENV CLANG_FORMAT=clang-format
-RUN ln -s /clang+llvm-5.0.0-linux-x86_64-ubuntu14.04/bin/clang-tidy /usr/local/bin/clang-tidy
-ENV CLANG_TIDY=clang-tidy
+FROM debian:10
 
+RUN apt-get update && apt-get install -y clang-format
```
Comment (jtattermusch) [inline tools/dockerfile/grpc_clang_format/Dockerfile]: freeze the version by adding `clang-format-7`?

File: tools/dockerfile/test/sanity/Dockerfile [hunk 4]
```diff
@@ -83,13 +82,16 @@ RUN apt-get update && apt-get install -y \
...
 
-RUN apt-get update && apt-get -y install wget xz-utils
-RUN wget http://releases.llvm.org/5.0.0/clang+llvm-5.0.0-linux-x86_64-ubuntu14.04.tar.xz
-RUN tar xf clang+llvm-5.0.0-linux-x86_64-ubuntu14.04.tar.xz
-RUN ln -s /clang+llvm-5.0.0-linux-x86_64-ubuntu14.04/bin/clang-format /usr/local/bin/clang-format
-ENV CLANG_FORMAT=clang-format
-RUN ln -s /clang+llvm-5.0.0-linux-x86_64-ubuntu14.04/bin/clang-tidy /usr/local/bin/clang-tidy
-ENV CLANG_TIDY=clang-tidy
+# Install clang & clang-format
+RUN apt-get install -y clang clang-format
```
Comment (jtattermusch) [inline tools/dockerfile/test/sanity/Dockerfile]: freeeze the version by adding `-7` version suffix?

**Pull request merged**